src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating()){
```

Comment:
Coding standard violation here w.r.t. spacing between `)` and `}`. Refer to our coding standards for the solution.